src/test/java/guitests/TagPersonGuiTest.java

```diff
@@ -0,0 +1,276 @@
+package guitests;
+
+import address.model.datatypes.AddressBook;
+import guitests.guihandles.EditPersonDialogHandle;
+import guitests.guihandles.PersonCardHandle;
+import guitests.guihandles.TagPersonDialogHandle;
+import org.junit.Test;
+
+import java.util.Arrays;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class TagPersonGuiTest extends GuiTestBase {
+    @Override
+    protected AddressBook getInitialData() {
+        return td.book;
+    }
+
+    @Test
+    public void tagPerson() {
```

Comment:
Various cases of adding one tag at a time.
1. Add to a person without tags.
2. Add to a person with existing tags.
3. Add tag to a person having the same tag already.